### ISSUE TITLE
composer network install fails v 0.20

### ISSUE THREAD

**Issue by devbnj** [hyperledger/composer#4337]
Trying to deploy a new Network through composer network install and it fails.
## CLI
composer network install --card PeerAdmin@hlfv1 --archiveFile mortgage@0.0.1.bna

## Error
Error: Error trying to start business network. Error: No valid responses from any peers.
Response from attempted peer comms was an error: Error: REQUEST_TIMEOUT

## Existing issues

- [ ] [Stack Overflow issues](http://stackoverflow.com/tags/hyperledger-composer)
- [x ] [GitHub Issues](https://github.com/hyperledger/composer/issues/3731)
- [ ] [Rocket Chat history](https://chat.hyperledger.org/channel/composer)

## Your Environment

* Version used: 0.20
* Environment name and version (e.g. Chrome 39, node.js 5.4): Node.js v8.11.3 Docker version 18.03.0-ce, build 0520e24
* Operating System and version (desktop or mobile):MacOS High Sierra v 10.13.6
* Link to your project:

**Comment by rajrawoor** [comment]
I get the following error when trying to start the network .
I am using Hyperledger fabric 1.2 and composer 0.20.


> composer network start -c PeerAdmin@byfn-network-org1 -n scmblockchain -V 0.0.3-deploy.0 -o endorsementPolicyFile=/tmp/composer/endorsement-policy.json -A alice -C alice/admin-pub.pem 


✖ Starting business network definition. This may take a minute...
Error: Error trying to start business network. Error: Failed to connect before the deadline
Command failed

@davidkel  .. Can you please suggest.. I have tried everything you recommended from issue #3731.

**Comment by 5idu** [comment]
I get the following error when trying to start the network .
Error: Error trying to start business network. Error: No valid responses from any peers.
Response from attempted peer comms was an error: Error: REQUEST_TIMEOUT

**Comment by mahoney1** [comment]
hello, please see the Composer knowledge wiki -> https://github.com/hyperledger/composer-knowledge-wiki/blob/latest/knowledge.md#information_source--composer-network-start-issues and scroll down (see REQUEST_TIMEOUT section - approx row 9 in that table) - you need to change the time out in both the YAML files (then use docker-compose stop / start for changes take effect) AND ALSO change the timeouts (as described) in the connection.json file of the card being used for the `composer network start` command sequence.

**Comment by devbnj** [comment]
@mahoney1 I did make the changes in docker-compose.yml in the peer section
      - CORE_VM_DOCKER_ATTACHSTDOUT=true
      - CORE_CHAINCODE_STARTUPTIMEOUT=1200s
      - CORE_CHAINCODE_EXECUTETIMEOUT=800s
I also changed the timers in the connection.json
    "x-commitTimeout": 1200,
    "version": "1.0.0",
    "client": {
        "organization": "Org1",
        "connection": {
            "timeout": {
                "peer": {
                    "endorser": "1200",
                    "eventHub": "1200",
                    "eventReg": "1200"
                },
                "orderer": "1200"
Its still the same error. 

Error: Error trying install business network. Error: No valid responses from any peers.
Response from attempted peer comms was an error: Error: Failed to connect before the deadline
Command failed

I have a fast network and a decent cpu.

**Comment by rajrawoor** [comment]
now i am getting an error  on the  
> composer identity request -c PeerAdmin@byfn-network-org1 -u admin -s adminpw -d alice

Error: failed to request identity. Error trying to enroll user and return certificates. Error: Calling enrollment endpoint failed with error [Error: connect ETIMEDOUT 192.168.1.159:7054]
Command failed

hyperledger fabric 1.2
composer v0.20
i have already verified (and verified) all the ca certs and the tls certs for all peers are copied correctly in the byfn-network-org1.json . 
Also, the TLS is enabled in the docker-composer-cli.yaml 

cat ./org1/*json
{
    "name": "byfn-network",
    "x-type": "hlfv1",
    "version": "1.0.0",
    "client": {
        "organization": "Org1",
        "connection": {
            "timeout": {
                "peer": {
                    "endorser": "1200",
                    "eventHub": "1200",
                    "eventReg": "1200"
                },
                "orderer": "1200"
            }
        }
    },
    "channels": {
        "mychannel": {
            "orderers": [
                "orderer.example.com"
            ],
            "peers": {
                "peer0.org1.example.com": {
                    "endorsingPeer": true,
                    "chaincodeQuery": true,
                    "eventSource": true
                },
                "peer1.org1.example.com": {
                    "endorsingPeer": true,
                    "chaincodeQuery": true,
                    "eventSource": true
                },
                "peer0.org2.example.com": {
                    "endorsingPeer": true,
                    "chaincodeQuery": true,
                    "eventSource": true
                },
                "peer1.org2.example.com": {
                    "endorsingPeer": true,
                    "chaincodeQuery": true,
                    "eventSource": true
                }
            }
        }
    },
    "organizations": {
        "Org1": {
            "mspid": "Org1MSP",
            "peers": [
                "peer0.org1.example.com",
                "peer1.org1.example.com"
            ],
            "certificateAuthorities": [
                "ca.org1.example.com"
            ]
        },
        "Org2": {
            "mspid": "Org2MSP",
            "peers": [
                "peer0.org2.example.com",
                "peer1.org2.example.com"
            ],
            "certificateAuthorities": [
                "ca.org2.example.com"
            ]
        }
    },
    "orderers": {
        "orderer.example.com": {
            "url": "grpcs://localhost:7050",
            "grpcOptions": {
                "ssl-target-name-override": "orderer.example.com"
            },
            "tlsCACerts": {
                "pem": "-----BEGIN CERTIFICATE-----\nMIICNjCCAdygAwIBAgIRAObpbDwgoENq9ae9xt2Y7iMwCgYIKoZIzj0EAwIwbDEL\nMAkGA1UEBhMCVVMxEzARBgNVBAgTCkNhbGlmb3JuaWExFjAUBgNVBAcTDVNhbiBG\ncmFuY2lzY28xFDASBgNVBAoTC2V4YW1wbGUuY29tMRowGAYDVQQDExF0bHNjYS5l\neGFtcGxlLmNvbTAeFw0xODA4MTYwMTI3MTRaFw0yODA4MTMwMTI3MTRaMGwxCzAJ\nBgNVBAYTAlVTMRMwEQYDVQQIEwpDYWxpZm9ybmlhMRYwFAYDVQQHEw1TYW4gRnJh\nbmNpc2NvMRQwEgYDVQQKEwtleGFtcGxlLmNvbTEaMBgGA1UEAxMRdGxzY2EuZXhh\nbXBsZS5jb20wWTATBgcqhkjOPQIBBggqhkjOPQMBBwNCAAQVSKzVu9jhELWQ83zU\nR2FoiYvVDfIPtoFRgWkwe08DTYH8hbav5IQm2BW9cKpZ681weJCvY40TVO/gwk7H\nk64ho18wXTAOBgNVHQ8BAf8EBAMCAaYwDwYDVR0lBAgwBgYEVR0lADAPBgNVHRMB\nAf8EBTADAQH/MCkGA1UdDgQiBCB4ezOoLazLorKZVaP13YOZtzxObPgXiPkQElmG\nsuJZWDAKBggqhkjOPQQDAgNIADBFAiEArtkBLDF23yKQ+uiC3dPQr3AQjyLDCXn6\nKmGQpnt/S8cCIBj0tQUH6eC4A9O9ESq2bg+z2aTz3KIU8iekQVJVw+gK\n-----END CERTIFICATE-----\n"
            }
        }
    },
    "peers": {
        "peer0.org1.example.com": {
            "url": "grpcs://localhost.com:7051",
            "grpcOptions": {
                "ssl-target-name-override": "peer0.org1.example.com"
            },
            "tlsCACerts": {
                "pem": "-----BEGIN CERTIFICATE-----\nMIICSjCCAfCgAwIBAgIRAJXaYLgBAwxVBW9CaV+CgyMwCgYIKoZIzj0EAwIwdjEL\nMAkGA1UEBhMCVVMxEzARBgNVBAgTCkNhbGlmb3JuaWExFjAUBgNVBAcTDVNhbiBG\ncmFuY2lzY28xGTAXBgNVBAoTEG9yZzEuZXhhbXBsZS5jb20xHzAdBgNVBAMTFnRs\nc2NhLm9yZzEuZXhhbXBsZS5jb20wHhcNMTgwODE2MDEyNzE0WhcNMjgwODEzMDEy\nNzE0WjB2MQswCQYDVQQGEwJVUzETMBEGA1UECBMKQ2FsaWZvcm5pYTEWMBQGA1UE\nBxMNU2FuIEZyYW5jaXNjbzEZMBcGA1UEChMQb3JnMS5leGFtcGxlLmNvbTEfMB0G\nA1UEAxMWdGxzY2Eub3JnMS5leGFtcGxlLmNvbTBZMBMGByqGSM49AgEGCCqGSM49\nAwEHA0IABJVVanFTaisufMlNtKJt/hcvFE2aB+3gY0P+3HX+1ePIlbmwh9P4pDhN\ndc+mX7g//gKVV+R7NQnDVdn4mdEasX6jXzBdMA4GA1UdDwEB/wQEAwIBpjAPBgNV\nHSUECDAGBgRVHSUAMA8GA1UdEwEB/wQFMAMBAf8wKQYDVR0OBCIEINdm7BDMYTV9\nRWcVHpgv7ghZ7OWj20Mes6Lie+himHw0MAoGCCqGSM49BAMCA0gAMEUCIQDXf1nh\nIvTzP+qwFnHa8iRvJz72NR8hO0FEKjf78yovYgIgerLVhjW5ayyFy4alXUrGL+oO\n5UrDMBF6mWjEpLHiP3k=\n-----END CERTIFICATE-----\n"
            }
        },
        "peer1.org1.example.com": {
            "url": "grpcs://peer1.org1.example.com:8051",
            "grpcOptions": {
                "ssl-target-name-override": "peer1.org1.example.com"
            },
            "tlsCACerts": {
                "pem": "-----BEGIN CERTIFICATE-----\nMIICSjCCAfCgAwIBAgIRAJXaYLgBAwxVBW9CaV+CgyMwCgYIKoZIzj0EAwIwdjEL\nMAkGA1UEBhMCVVMxEzARBgNVBAgTCkNhbGlmb3JuaWExFjAUBgNVBAcTDVNhbiBG\ncmFuY2lzY28xGTAXBgNVBAoTEG9yZzEuZXhhbXBsZS5jb20xHzAdBgNVBAMTFnRs\nc2NhLm9yZzEuZXhhbXBsZS5jb20wHhcNMTgwODE2MDEyNzE0WhcNMjgwODEzMDEy\nNzE0WjB2MQswCQYDVQQGEwJVUzETMBEGA1UECBMKQ2FsaWZvcm5pYTEWMBQGA1UE\nBxMNU2FuIEZyYW5jaXNjbzEZMBcGA1UEChMQb3JnMS5leGFtcGxlLmNvbTEfMB0G\nA1UEAxMWdGxzY2Eub3JnMS5leGFtcGxlLmNvbTBZMBMGByqGSM49AgEGCCqGSM49\nAwEHA0IABJVVanFTaisufMlNtKJt/hcvFE2aB+3gY0P+3HX+1ePIlbmwh9P4pDhN\ndc+mX7g//gKVV+R7NQnDVdn4mdEasX6jXzBdMA4GA1UdDwEB/wQEAwIBpjAPBgNV\nHSUECDAGBgRVHSUAMA8GA1UdEwEB/wQFMAMBAf8wKQYDVR0OBCIEINdm7BDMYTV9\nRWcVHpgv7ghZ7OWj20Mes6Lie+himHw0MAoGCCqGSM49BAMCA0gAMEUCIQDXf1nh\nIvTzP+qwFnHa8iRvJz72NR8hO0FEKjf78yovYgIgerLVhjW5ayyFy4alXUrGL+oO\n5UrDMBF6mWjEpLHiP3k=\n-----END CERTIFICATE-----\n"
            }
        },
        "peer0.org2.example.com": {
            "url": "grpcs://localhost.com:9051",
            "grpcOptions": {
                "ssl-target-name-override": "peer0.org2.example.com"
            },
            "tlsCACerts": {
                "pem": "-----BEGIN CERTIFICATE-----\nMIICSjCCAfCgAwIBAgIRAMZ3rMvIJP+ebMYvxTsrqEswCgYIKoZIzj0EAwIwdjEL\nMAkGA1UEBhMCVVMxEzARBgNVBAgTCkNhbGlmb3JuaWExFjAUBgNVBAcTDVNhbiBG\ncmFuY2lzY28xGTAXBgNVBAoTEG9yZzIuZXhhbXBsZS5jb20xHzAdBgNVBAMTFnRs\nc2NhLm9yZzIuZXhhbXBsZS5jb20wHhcNMTgwODE2MDEyNzE0WhcNMjgwODEzMDEy\nNzE0WjB2MQswCQYDVQQGEwJVUzETMBEGA1UECBMKQ2FsaWZvcm5pYTEWMBQGA1UE\nBxMNU2FuIEZyYW5jaXNjbzEZMBcGA1UEChMQb3JnMi5leGFtcGxlLmNvbTEfMB0G\nA1UEAxMWdGxzY2Eub3JnMi5leGFtcGxlLmNvbTBZMBMGByqGSM49AgEGCCqGSM49\nAwEHA0IABD9NNQrngFbpYV7CqOMk1yDcVqo5wl4XBiSQ9vVCEcFa5kTRUYX5zJcb\nsp9o0CCyDHOyY3aq9RDUnwyP+0W00qCjXzBdMA4GA1UdDwEB/wQEAwIBpjAPBgNV\nHSUECDAGBgRVHSUAMA8GA1UdEwEB/wQFMAMBAf8wKQYDVR0OBCIEIGqaM0oY+4bk\nMAyHivCN12AAq/52PnpO4cUvQCSSy1XZMAoGCCqGSM49BAMCA0gAMEUCIQCsMWyw\nz/k9prAf043Kv/de0/avLm7EXvNDx1kaXQc9CwIgYuuCxvz8z1BdV7Ib/+knTcnX\n9qkKVLGC/L/opKL/mYM=\n-----END CERTIFICATE-----\n"
            }
        },
        "peer1.org2.example.com": {
            "url": "grpcs://localhost.com:10051",
            "grpcOptions": {
                "ssl-target-name-override": "peer1.org2.example.com"
            },
            "tlsCACerts": {
                "pem": "-----BEGIN CERTIFICATE-----\nMIICSjCCAfCgAwIBAgIRAMZ3rMvIJP+ebMYvxTsrqEswCgYIKoZIzj0EAwIwdjEL\nMAkGA1UEBhMCVVMxEzARBgNVBAgTCkNhbGlmb3JuaWExFjAUBgNVBAcTDVNhbiBG\ncmFuY2lzY28xGTAXBgNVBAoTEG9yZzIuZXhhbXBsZS5jb20xHzAdBgNVBAMTFnRs\nc2NhLm9yZzIuZXhhbXBsZS5jb20wHhcNMTgwODE2MDEyNzE0WhcNMjgwODEzMDEy\nNzE0WjB2MQswCQYDVQQGEwJVUzETMBEGA1UECBMKQ2FsaWZvcm5pYTEWMBQGA1UE\nBxMNU2FuIEZyYW5jaXNjbzEZMBcGA1UEChMQb3JnMi5leGFtcGxlLmNvbTEfMB0G\nA1UEAxMWdGxzY2Eub3JnMi5leGFtcGxlLmNvbTBZMBMGByqGSM49AgEGCCqGSM49\nAwEHA0IABD9NNQrngFbpYV7CqOMk1yDcVqo5wl4XBiSQ9vVCEcFa5kTRUYX5zJcb\nsp9o0CCyDHOyY3aq9RDUnwyP+0W00qCjXzBdMA4GA1UdDwEB/wQEAwIBpjAPBgNV\nHSUECDAGBgRVHSUAMA8GA1UdEwEB/wQFMAMBAf8wKQYDVR0OBCIEIGqaM0oY+4bk\nMAyHivCN12AAq/52PnpO4cUvQCSSy1XZMAoGCCqGSM49BAMCA0gAMEUCIQCsMWyw\nz/k9prAf043Kv/de0/avLm7EXvNDx1kaXQc9CwIgYuuCxvz8z1BdV7Ib/+knTcnX\n9qkKVLGC/L/opKL/mYM=\n-----END CERTIFICATE-----\n"
            }
        }
    },
    "certificateAuthorities": {
        "ca.org1.example.com": {
            "url": "https://localhost:7054",
            "caName": "ca-org1"
        },
        "ca.org2.example.com": {
            "url": "https://localhost:8054",
            "caName": "ca-org2"
        }
    }
}

**Comment by mahoney1** [comment]
@rajrawoor  the answer to your issue is most likely a docker networking issue (your connection profile has all localhost defs, but the error from docker is a refusal on a port bound to a different IP address) - this [Stack Overflow](https://stackoverflow.com/questions/48416805/calling-enrollment-endpoint-failed-with-error-error-connect-econnrefused) has  a number of possible answers. I would suggest if you want to take it further - to post a question on Stack Overflow (the question should be of this form -> https://github.com/hyperledger/composer-knowledge-wiki/blob/latest/knowledge.md#issue)

**Comment by mahoney1** [comment]
@devbnj  - you didn't say if you had done a `docker-compose stop` and start for changes to take effect. I suggest to do a teardown of your local dev environment, remove the containers (keep your mortgage BNA file) and start your Fabric with the docker-compose up command. remove your old cards in $HOME/.composer ; then run ./createPeerAdminCard.sh again (from dev-servers directory etc) to create PeerAdmin card,  and edit its connection.json to match the timeouts you've got (as shown above). Then do a composer network install, wait 5 secs, then do the composer network start sequence. Any more issues, please raise on Stack Overflow, where answers can be provided. Closing this issue.

**Comment by devbnj** [comment]
@mahoney1 - Paul, it simply work. Not on MacOS. I guess I will revert back to an older version where things are working. I am sure a lot of folks will run into this problem. In any case, deleting everything is not going to help customers who (like me) will be scratching their heads on why they have to delete everything in minor version upgrades.

**Comment by mahoney1** [comment]
hi @devbnj ok thanks for letting me know -- I agree, shouldn't have to delete everything to make it work. The error is a networking issue (its on localhost, but something isn't right). I might suggest to see what happens if you put the 172.x ip address of your m/c as an additional IP alias (eg localhost 127.0.0.1 172.16.0.9 ) in your /etc/hosts file and see if (on v0.20 env) it made any difference to your `composer network install`. I assume you've installed a Fabric 1.2 environment (using the new Fabric dev servers scripts) for 0.20.0 of Composer. Anyway, hope this helps.

**Comment by devbnj** [comment]
Hi @mahoney1 , just wondering if its not working with any of these - how is it working at your developers end (desktop or cloud instance)? Something seems to be missing. I cleaned up the environment, but I can't clean up my machine any more - as I use it for other purposes also. It simply refuses to connect. Incidentally, I have downloaded the Fabric 12 samples (not composer based), changed the IP address from localhost to the docker IP and the node.js programs work pretty well. I guess it is localhost - can I make similar changes to composer? I could not detect the place to change in the composer source code. Thought I'd ask you.
